File: app/src/main/java/com/my/test2/TestAdapter.java
package com.my.test2;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.my.test2.databinding.ItemTestBinding;

import java.util.List;

/**
 * @author : yx^_^
 * @e-mail : 565749553@qq.com
 * @date : 2020/3/24 14:38
 * @desc : better than before
 */
public class TestAdapter extends BaseAdapter<String,ItemTestBinding> {
    public TestAdapter(Context context, List <String>mDatas) {
        super(context, mDatas);
    }

    @Override
    protected void onBindHolder(ViewHolder<ItemTestBinding> holder, int position) {
            holder.getBinding().tx.setText(mDatas.get(position));
    }

    @Override
    protected ItemTestBinding setBinding(LayoutInflater inflater, ViewGroup parent) {
        return ItemTestBinding.inflate(inflater,parent,false);
    }
}

File: app/src/main/java/com/my/test2/kotlin/TestAdapter.kt
package com.dsf.vms

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import com.dsf.dsfwarehouse.base.BaseHaveFooterAdapter
import com.dsf.vms.databinding.AdapterFootTestBinding
import com.dsf.vms.databinding.AdapterItemTestBinding

class  TestAdapter(context: Context, mDatas: List<String>?) :
    BaseHaveFooterAdapter<String, AdapterItemTestBinding, AdapterFootTestBinding>(context, mDatas) {
    override fun onBindHolder(holder: ViewHolder<AdapterFootTestBinding>, position: Int) {
        val binding=holder.binding as AdapterItemTestBinding;
        binding.tv.setText(mDatas?.get(position))
        holder.setClick(binding.root)

    }

    override fun onBindFooterHolder(holder: ViewHolder<AdapterItemTestBinding>, position: Int) {
        val binding=holder.binding as AdapterFootTestBinding
        holder.setOnFooterClick(binding.foot)
    }

    override fun setBinding(inflater: LayoutInflater?, parent: ViewGroup?): AdapterItemTestBinding {
        return AdapterItemTestBinding.inflate(inflater!!,parent,false);
    }

    override fun setFootBinding(
        inflater: LayoutInflater?,
        parent: ViewGroup?
    ): AdapterFootTestBinding {
        return AdapterFootTestBinding.inflate(inflater!!,parent,false);
    }

}